### ISSUE TITLE
Not functional in Fabric 1.14.4

### ISSUE THREAD

**Issue by jivvy** [gbl/EasierCrafting#14]
On world load:

---- Minecraft Crash Report ----
// This is a token for 1 free hug. Redeem at your nearest Mojangsta: [~~HUG~~]

Time: 7/27/19 1:22 PM
Description: Unexpected error

java.lang.NullPointerException: Unexpected error
	at de.guntram.mcmod.fabrictools.KeyBindingManager.processKeyBinds(KeyBindingManager.java:27)
	at net.minecraft.class_310.handler$onProcessKeybinds$zcf000(class_310.java:3972)
	at net.minecraft.class_310.method_1508(class_310.java)
	at net.minecraft.class_310.method_1574(class_310.java:1411)
	at net.minecraft.class_310.method_1523(class_310.java:925)
	at net.minecraft.class_310.method_1514(class_310.java:410)
	at net.minecraft.client.main.Main.main(Main.java:155)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:170)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:129)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at de.guntram.mcmod.fabrictools.KeyBindingManager.processKeyBinds(KeyBindingManager.java:27)
	at net.minecraft.class_310.handler$onProcessKeybinds$zcf000(class_310.java:3972)
	at net.minecraft.class_310.method_1508(class_310.java)

-- Affected level --
Details:
	All players: 1 total; [class_746['Jivvyn'/124963, l='MpServer', x=-236.82, y=14.00, z=-2135.06]]
	Chunk stats: Client Chunk Cache: 625, 361
	Level dimension: minecraft:overworld
	Level name: MpServer
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: {}
	Level spawn location: World: (-224,70,-239), Chunk: (at 0,4,1 in -14,-15; contains blocks -224,0,-240 to -209,255,-225), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
	Level time: 99985 game time, 26232990 day time
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
	Server brand: Paper
	Server type: Non-integrated multiplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:421)
	at net.minecraft.class_310.method_1587(class_310.java:1923)
	at net.minecraft.class_310.method_1514(class_310.java:430)
	at net.minecraft.client.main.Main.main(Main.java:155)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:170)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:129)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)

-- System Details --
Details:
	Minecraft Version: 1.14.4
	Minecraft Version ID: 1.14.4
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 1499182296 bytes (1429 MB) / 2080374784 bytes (1984 MB) up to 2147483648 bytes (2048 MB)
	CPUs: 8
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		easiercrafting: EasierCrafting 1.14-fabric0.2.7-1.3.0
		fabric: Fabric API 0.3.0+build.200
		fabric-api-base: fabric-api-base 0.1.0+5914746342
		fabric-biomes-v1: fabric-biomes-v1 0.1.0+896c7fbb42
		fabric-commands-v0: fabric-commands-v0 0.1.1+25fd0c5242
		fabric-containers-v0: fabric-containers-v0 0.1.2+25fd0c5242
		fabric-content-registries-v0: fabric-content-registries-v0 0.1.1+05f11b7542
		fabric-crash-report-info-v1: fabric-crash-report-info-v1 0.1.0+5914746342
		fabric-events-interaction-v0: fabric-events-interaction-v0 0.1.0+ea100f6142
		fabric-events-lifecycle-v0: fabric-events-lifecycle-v0 0.1.0+5914746342
		fabric-item-groups-v0: fabric-item-groups-v0 0.1.0+1e69a0a542
		fabric-keybindings-v0: fabric-keybindings-v0 0.1.1+7dfef87c42
		fabric-loot-tables-v1: fabric-loot-tables-v1 0.1.0+43028fa642
		fabric-mining-levels-v0: fabric-mining-levels-v0 0.1.0+ab421b9c42
		fabric-models-v0: fabric-models-v0 0.1.0+5914746342
		fabric-networking-blockentity-v0: fabric-networking-blockentity-v0 0.1.1+25fd0c5242
		fabric-networking-v0: fabric-networking-v0 0.1.2+ea100f6142
		fabric-object-builders-v0: fabric-object-builders-v0 0.1.1+592d1d2242
		fabric-registry-sync-v0: fabric-registry-sync-v0 0.2.2+ea100f6142
		fabric-renderer-api-v1: fabric-renderer-api-v1 0.1.1+9486cfbf42
		fabric-renderer-indigo: fabric-renderer-indigo 0.1.9+9486cfbf42
		fabric-rendering-data-attachment-v1: fabric-rendering-data-attachment-v1 0.1.0+02a46d5b42
		fabric-rendering-fluids-v1: fabric-rendering-fluids-v1 0.1.0+dc4c57c242
		fabric-rendering-v0: fabric-rendering-v0 0.1.1+6dad974e42
		fabric-resource-loader-v0: fabric-resource-loader-v0 0.1.1+7495ea2842
		fabric-tag-extensions-v0: fabric-tag-extensions-v0 0.1.1+c189dc5c42
		fabric-textures-v0: fabric-textures-v0 0.1.4+792c1c3542
		fabricloader: Fabric Loader 0.4.8+build.158
		gbfabrictools: GBfabrictools 1.0.0
		malilib: MaLiLib 0.10.0-dev.20
	Launched Version: fabric-loader-0.4.8+build.158-1.14.4
	LWJGL: 3.2.2 build 10
	OpenGL: GeForce GTX 1050 Ti/PCIe/SSE2 GL version 4.6.0 NVIDIA 399.24, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Resource Packs: vanilla, file/1.14.3 Vanilla Tweaks.zip
	Current Language: English (US)
	CPU: 8x AMD Ryzen 5 1500X Quad-Core Processor

**Comment by Azutte** [comment]
same 
```
Unreported exception thrown!
java.lang.NullPointerException
	at de.guntram.mcmod.fabrictools.KeyBindingManager.processKeyBinds(KeyBindingManager.java:27)
	at net.minecraft.class_310.handler$onProcessKeybinds$zch000(class_310.java:3972)
	at net.minecraft.class_310.method_1508(class_310.java)
	at net.minecraft.class_310.method_1574(class_310.java:1411)
	at net.minecraft.class_310.method_1523(class_310.java:925)
	at net.minecraft.class_310.method_1514(class_310.java:410)
	at net.minecraft.client.main.Main.main(Main.java:155)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:178)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:131)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
16:28:11.598
launcher
main

---- Minecraft Crash Report ----
// I blame Dinnerbone.

Time: 9/11/19 4:28 PM
Description: Unexpected error

java.lang.NullPointerException: Unexpected error
	at de.guntram.mcmod.fabrictools.KeyBindingManager.processKeyBinds(KeyBindingManager.java:27)
	at net.minecraft.class_310.handler$onProcessKeybinds$zch000(class_310.java:3972)
	at net.minecraft.class_310.method_1508(class_310.java)
	at net.minecraft.class_310.method_1574(class_310.java:1411)
	at net.minecraft.class_310.method_1523(class_310.java:925)
	at net.minecraft.class_310.method_1514(class_310.java:410)
	at net.minecraft.client.main.Main.main(Main.java:155)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:178)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:131)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at de.guntram.mcmod.fabrictools.KeyBindingManager.processKeyBinds(KeyBindingManager.java:27)
	at net.minecraft.class_310.handler$onProcessKeybinds$zch000(class_310.java:3972)
	at net.minecraft.class_310.method_1508(class_310.java)

```

**Comment by Paradoxis** [comment]
@Azutte @Jivvy the issue is caused by the fact that a library is not bundled with the mod, nor is it explained that it needs to be installed for the mod to work. You'll need to compile [this](https://github.com/gbl/GBfabrictools) library written by @gbl and copy the jar to your mods file.

You can compile it by using `gradle build` and copying the 'GBfabrictools-1.0.1.jar' file from the `builds/libs` folder to your mods folder. Alternatively,  just download [this](https://mega.nz/#!S5JSTAxT!QOYmB2eTEm8gYE6_1SWY9qwXiHBfowLAhI2jEhqYo9w) precompiled jar file and put it in your mods folder.

@gbl it might be a good idea to test your mod on a clean install before releasing it in the future, saves a lot of people headaches :)

**Comment by gbl** [comment]
Actually the problem is/was something else; my fabric tools library included a keybind manager, but if no mod used said keybind manager, it would crash. Is fixed in new versions. (It's also fixed in GBfabrictools-1.0.1, but it seems I didn't bundle that version with the mod). If any other of my mods is installed, there shouldn't be a problem.
